Check competition dates for upcoming comps while banning

danieljames-dj opened the pull request:
This was an edge case noticed by Shain and the WIC also has an urgent case which is blocked because of this

gregorbg commented:
Could you please explain this edge case?

danieljames-dj commented:
> Could you please explain this edge case?

If we are trying to ban a competitor for 1 year, and if the competitor has a registration after 1 year, still they should be allowed to be banned because their upcoming competition is not within the ban window.

gregorbg commented on spec/controllers/api/v0/user_roles_controller_spec.rb:
```diff
@@ -125,6 +125,15 @@
         expect(response_json["error"]).to eq "The user has upcoming competitions: #{upcoming_comps_for_user.join(', ')}. Before banning the user, make sure their registrations are deleted."
       end
 
+      it 'can ban a user if the user\'s upcoming competitions are after end date' do
```
You can make this double quotes, then you don't have to mask out the apostrophe in `don't` 
```suggestion
      it "can ban a user if the user's upcoming competitions are after end date" do
```

danieljames-dj replied:
Done

gregorbg reviewed (commented):
You need to also fix the other test case which is failing ("cannot ban a user if the user have any upcoming competitions"), otherwise fine to merge :+1:

danieljames-dj commented:
> You need to also fix the other test case which is failing ("cannot ban a user if the user have any upcoming competitions"), otherwise fine to merge 👍

Fixed it, changed the code a bit to fix the test.
